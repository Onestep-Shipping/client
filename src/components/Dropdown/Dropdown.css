.login-container {
  position: relative;
  display: flex;
  flex-direction: row-reverse;
  align-items: flex-end;
}

.user-button {
  background-color: transparent;
  color: white;
  font-size: 16px;
  border-radius: 2px;
  border: none;
  outline: none;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-left: 0px;
  margin-top: 5px;
  user-select: none;
}

.dropdown-menu {
  list-style-type: none;
  margin: 0;
  padding: 0;
  top: 40px;
  right: 15px;
  background-color: white;
  position: absolute;

  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  z-index: 1;
  margin-top: 5px;
  border: 1px solid #6d6d6d
}

.dropdown-menu-item {
  padding: 8px 16px;
  border-bottom: 1px solid #e5e5e5;
  cursor: pointer;
  line-height: 25px;
}

.dropdown-menu-item:last-child {
  border-bottom: none;
}
.dropdown-menu-item:hover {
  background-color: #e5e5e5;
  color: black;
}