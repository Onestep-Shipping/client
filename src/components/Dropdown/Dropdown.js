import React, {useState, useEffect, useRef, useCallback} from 'react';
import './Dropdown.css';
import PropTypes from 'prop-types';
import AuthModal from '../AuthModal/AuthModal.js';

const Dropdown = (props) => {
  const node = useRef();
  const [open, setOpen] = useState(false);
  const [isModalOpen, setModalOpen] = useState(false);

  const {type = '', content, options = [], onChange, isLoggedIn = false, isAlwaysVisible = true} = props;

  const openModal = () => {
    setModalOpen(true);
  }
 
  const closeModal = () => {
    setModalOpen(false);
  }

  const measuredRef = useCallback((e) => {
    if (node.current && !node.current.contains(e.target)) {
      setOpen(false);
    }
  }, []);

  const handleChange = selectedValue => {
    onChange(selectedValue);
    setOpen(false);
  };

  useEffect(() => {
    document.addEventListener('mousedown', measuredRef);

    return () => {
      document.removeEventListener('mousedown', measuredRef);
    };
  }, [measuredRef]);

  const handleAuth = () => {
    if (isLoggedIn) {
      setOpen(!open);
    } else {
      openModal();
    }
  }

  if (!isAlwaysVisible) {
    return <div />;
  }

  return (
    <div ref={node} className="login-container">
      <button className="user-button" onClick={handleAuth}>
        {content}
      </button>
      {open && (
        <ul className="dropdown-menu">
          {options.map((opt, ind) => (
            <li className="dropdown-menu-item" key={ind} onClick={() => handleChange(opt)}>
              {opt}
            </li>
          ))}
        </ul>
      )}
      <AuthModal
        isModalOpen={isModalOpen}
        closeModal={closeModal}
        type={type} />
    </div>
  );
};

export default Dropdown;

Dropdown.propTypes = {
  type: PropTypes.string,
  content: PropTypes.element,
  options: PropTypes.array,
  onChange: PropTypes.func,
  isLoggedIn: PropTypes.bool,
  isAlwaysVisible: PropTypes.bool
};

