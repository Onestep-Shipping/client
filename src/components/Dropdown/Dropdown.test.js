import Dropdown from './Dropdown';
import React from 'react';
import { shallow } from 'enzyme';

let wrapper = shallow(
  <Dropdown
    content={<div />}
    type={'login'}
    onChange={() => {}}
    options={[]}
    isLoggedIn={false} 
  />
);

describe('Dropdown', () => {
  it('should render the Dropdown Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
});