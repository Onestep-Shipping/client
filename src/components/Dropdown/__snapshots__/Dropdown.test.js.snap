// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render the Dropdown Component correctly 1`] = `
<div
  className="login-container"
>
  <button
    className="user-button"
    onClick={[Function]}
  >
    <div />
  </button>
  <AuthModal
    closeModal={[Function]}
    isModalOpen={false}
    type="login"
  />
</div>
`;
